tests/xdg_popup_v6.cpp

```diff
@@ -0,0 +1,270 @@
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "helpers.h"
+#include "in_process_server.h"
+#include "xdg_shell_v6.h"
+
+#include <gmock/gmock.h>
+
+#include <experimental/optional>
+
+using namespace testing;
+
+int const window_width = 200, window_height = 300;
+int const popup_width = window_width - 100, popup_height = window_height - 120;
+
+class XdgPopupV6TestBase : public wlcs::StartedInProcessServer
+{
+public:
+    static int const window_x = 500, window_y = 500;
+
+    XdgPopupV6TestBase()
+        : client{the_server()},
+          surface{client},
+          xdg_surface{client, surface},
+          toplevel{xdg_surface},
+          positioner{client}
+    {
+        surface.attach_buffer(window_width, window_height);
+        bool surface_rendered{false};
+        surface.add_frame_callback([&surface_rendered](auto) { surface_rendered = true; });
+        wl_surface_commit(surface);
+        client.dispatch_until([&surface_rendered]() { return surface_rendered; });
+        the_server().move_surface_to(surface, window_x, window_y);
+    }
+
+    void map_popup()
+    {
+        popup_surface.emplace(client);
+        popup_xdg_surface.emplace(client, popup_surface.value());
+        popup.emplace(popup_xdg_surface.value(), xdg_surface, positioner);
+
+        popup_xdg_surface.value().add_configure_notification([&](uint32_t serial)
+            {
+                zxdg_surface_v6_ack_configure(popup_xdg_surface.value(), serial);
+                popup_surface_configure_count++;
+            });
+
+        popup.value().add_configure_notification([this](int32_t x, int32_t y, int32_t width, int32_t height)
+            {
+                state = wlcs::XdgPopupV6::State{x, y, width, height};
+            });
+
+        wl_surface_commit(popup_surface.value());
+        dispatch_until_popup_configure();
+        popup_surface.value().attach_buffer(popup_width, popup_height);
+        bool surface_rendered{false};
+        popup_surface.value().add_frame_callback([&surface_rendered](auto) { surface_rendered = true; });
+        wl_surface_commit(popup_surface.value());
+        client.dispatch_until([&surface_rendered]() { return surface_rendered; });
+    }
+
+    void dispatch_until_popup_configure()
+    {
+        client.dispatch_until(
+            [prev_count = popup_surface_configure_count, &current_count = popup_surface_configure_count]()
+            {
+                return current_count > prev_count;
+            });
+    }
+
+    std::experimental::optional<std::pair<int, int>> get_popup_position()
+    {
+        // state is not set, as WLCS window manager does not send the proper configure events.
+        // EXPECT_THAT(state.x, Eq(0));
+        // EXPECT_THAT(state.y, Eq(0));
+        // EXPECT_THAT(state.width, Eq(popup_width));
+        // EXPECT_THAT(state.height, Eq(popup_height));
```

Comment:
Is there a "WLCS window manager"? Do you mean the one use by miral-integration?

If so, this is a bug? Do we have an issue/pr?

Reply:
Ah, that was written before the switch to a miral window manager. I'll see check if we're getting the right events now.

Reply:
Resolved